worker_types/all-worker-types/main.go

```diff
@@ -105,10 +105,9 @@ func (wtf *WorkerTypeFetcher) fetch(workerType string) (err error) {
 		return
 	}
 	defer func() {
+		err2 := resp.Body.Close()
```

Comment:
What's the logic here? Not shadowing `err`?

Reply:
`err` is returned by the function. If an error has already occurred before the defer function is called (`err != nil`) then we should return that existing error (`err`). However, if `err == nil` (i.e. no error has occurred prior to calling the defer function) but `err2 != nil` (i.e. closing the response body causes an error) then we should return `err2` instead. Whatever we set `err` to will be what gets returned, so setting `err` to `err2` in this case means that we will return the error from closing the response body if one happens, but previously no error had occurred. If an error had already occurred, and we fail to close the response body, we'll return the original error, not the error about closing the response body.